src/Shared/src/PropertyHelper/PropertyHelper.cs

```diff
@@ -420,18 +421,17 @@ private static PropertyHelper CreateInstance(PropertyInfo property)
             ConcurrentDictionary<Type, PropertyHelper[]> allPropertiesCache,
             ConcurrentDictionary<Type, PropertyHelper[]> visiblePropertiesCache)
         {
-            PropertyHelper[] result;
-            if (visiblePropertiesCache.TryGetValue(type, out result))
+            if (visiblePropertiesCache.TryGetValue(type, out PropertyHelper[] result))
             {
                 return result;
             }
 
             // The simple and common case, this is normal POCO object - no need to allocate.
             var allPropertiesDefinedOnType = true;
             var allProperties = GetProperties(type, createPropertyHelper, allPropertiesCache);
-            foreach (var propertyHelper in allProperties)
+            for (var i = 0; i < allProperties.Length; i++)
```

Comment:
The C#-compiler transforms the `foreach`-loop to the `for`-loop when foreaching over an array.

So here it is more or less a matter of taste / style, but I'll keep to `foreach`.